File: ultimate/MiddlewareResponse.java
package ultimate;

public class MiddlewareResponse {

    private String message;
    private boolean hasSucceeded;

    public MiddlewareResponse(String message, boolean hasSucceeded) {
        this.message = message;
        this.hasSucceeded = hasSucceeded;
    }

    public String getMessage() {
        return message;
    }

    public boolean isHasSucceeded() {
        return hasSucceeded;
    }
}

File: typescript/patterns/ultimate-design-pattern-challenge/middleware-response.ts
class MiddlewareResponse {
  private message: string;
  private hasSucceeded: boolean;

  constructor(message: string, hasSucceeded: boolean) {
    this.message = message;
    this.hasSucceeded = hasSucceeded;
  }

  getMessage(): string {
    return this.message;
  }

  getHasSucceeded(): boolean {
    return this.hasSucceeded;
  }
}
